### PR TITLE
Rewrite the introduction of the Record Layer section.

### PR DESCRIPTION
**Pull request by ol-anssi**
As proposed on the mailing list, I propose some rewrites of the rules regarding record splitting/coalescing/merging. Some text has been moved (and more could be). The idea is to get all the relevant information in the same place.

I did not add constraints on HS splitting/merging (beyond those currently in place).

### CODE REVIEW

File: draft-ietf-tls-tls13.md [hunk 1]
```diff
@@ -3202,21 +3198,52 @@ with an "unexpected_message" alert.  New record content type values are
...
-The record layer fragments information blocks into TLSPlaintext records
-carrying data in chunks of 2^14 bytes or less. Message boundaries are
-not preserved in the record layer (i.e., multiple messages of the same
-ContentType MAY be coalesced into a single TLSPlaintext record, or a single
-message MAY be fragmented across several records).
-Alert messages ({{alert-protocol}}) MUST NOT be fragmented across records.
+The record layer fragments information blocks into TLSPlaintext
+records carrying data in chunks of 2^14 bytes or less. Message
+boundaries are handled differently depending on the underlying
+ContentType and MUST following rules.
```
Comment (martinthomson) [inline draft-ietf-tls-tls13.md]: This is a non-sentence.  You could stop after ContentType.

File: draft-ietf-tls-tls13.md [hunk 2]
```diff
@@ -3202,21 +3198,52 @@ with an "unexpected_message" alert.  New record content type values are
...
+
+Alert messages ({{alert-protocol}}) MUST NOT be fragmented across
+records and multiple Alert messages MUST NOT be coalesced into a
+single TLSPlaintext record. In other words, a record with an Alert
+type MUST contain exactly one message. This is the default behaviour
+to be adopted by new record content type.
```
Comment (martinthomson) [inline draft-ietf-tls-tls13.md]: "types" plural.

File: draft-ietf-tls-tls13.md [hunk 3]
```diff
@@ -3202,21 +3198,52 @@ with an "unexpected_message" alert.  New record content type values are
...
+
+Application Data messages are carried by the record layer and are
+fragmented and encrypted as described below. The messages are treated
+as transparent data to the record layer. Zero-length
```
Comment (martinthomson) [inline draft-ietf-tls-tls13.md]: not your text: but application data is transparent to TLS

File: draft-ietf-tls-tls13.md [hunk 4]
```diff
@@ -3202,21 +3198,52 @@ with an "unexpected_message" alert.  New record content type values are
...
+fragments of Application Data MAY be sent as they are potentially
+useful as a traffic analysis countermeasure.
+
+Since Handshake messages can be larger than a record content, a single
```
Comment (martinthomson) [inline draft-ietf-tls-tls13.md]: larger than a single record (drop content)

File: draft-ietf-tls-tls13.md [hunk 5]
```diff
@@ -3202,21 +3198,52 @@ with an "unexpected_message" alert.  New record content type values are
...
+Handshake message MAY be fragmented across several records. Moreover,
```
Comment (martinthomson) [inline draft-ietf-tls-tls13.md]: delete "Moreover"

File: draft-ietf-tls-tls13.md [hunk 6]
```diff
@@ -3202,21 +3198,52 @@ with an "unexpected_message" alert.  New record content type values are
...
+- Implementations MUST NOT send zero-length fragments of Handshake
+  types, even if those fragments contain padding.
+
+- Handshake messages MUST NOT be interleaved with other record
+  types. That is, if a message is split over two or more handshake
+  records, there MUST NOT be any other records between them.
+
+RFC EDITOR: The following point could be completely moved from the 4.6
+section if needed. This way, all the relevant information concerning
+record splitting/coalescing/interleaving would be in the same place.
```
Comment (martinthomson) [inline draft-ietf-tls-tls13.md]: Don't instruct the RFC editor here, just make the change you think best.  FWIW, it is good to repeat this here.

File: draft-ietf-tls-tls13.md [hunk 7]
```diff
@@ -3202,21 +3198,52 @@ with an "unexpected_message" alert.  New record content type values are
...
+ContentType and MUST following rules.
+
+Alert messages ({{alert-protocol}}) MUST NOT be fragmented across
+records and multiple Alert messages MUST NOT be coalesced into a
+single TLSPlaintext record. In other words, a record with an Alert
+type MUST contain exactly one message. This is the default behaviour
+to be adopted by new record content type.
+
+Application Data messages are carried by the record layer and are
+fragmented and encrypted as described below. The messages are treated
```
Comment (martinthomson) [inline draft-ietf-tls-tls13.md]: I don't know where *below* is now.  I think that you could state this more succinctly.

> Application Data messages contain data that is opaque to TLS.  Application Data messages are always protected.  Zero-length fragments of Application Data MAY [...]

**Comment by ekr** [comment]
I merged this and made some editorial revisions. @ol-anssi please provide a PR to add yourself to the contributors list.

**Pull request closed without merging**